demos/dqx_streaming_demo_diy.py

```diff
@@ -0,0 +1,249 @@
+      column: trip_distance
+  name: trip_distance_is_null
+  criticality: error
+""")
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Inputs
+# MAGIC - Checkpoint and table locations are set via widgets
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC
+# MAGIC <h3>Checkpoint and Table Location Options</h3>
+# MAGIC <ul>
+# MAGIC   <li>
+# MAGIC     <b>Checkpoint Location</b>:
+# MAGIC     <ul>
+# MAGIC       <li>Local path (e.g., <code>/tmp/checkpoints/...</code>)</li>
+# MAGIC       <li>Workspace path (e.g., <code>/dbfs/mnt/...</code>)</li>
+# MAGIC       <li>Unity Catalog (UC) volume path (e.g., <code>/Volumes/catalog/schema/volume/...</code>)</li>
+# MAGIC     </ul>
+# MAGIC   </li>
+# MAGIC   <li>
+# MAGIC     <b>Silver and Quarantine Table</b>:
+# MAGIC     <ul>
+# MAGIC       <li>Delta table path (e.g., <code>/mnt/delta/silver_table</code>)</li>
+# MAGIC       <li>Unity Catalog table (e.g., <code>catalog.schema.table</code>)</li>
+# MAGIC     </ul>
+# MAGIC   </li>
+# MAGIC </ul>
+# MAGIC
+
+# COMMAND ----------
+
+dbutils.widgets.text("silver_checkpoint", "/tmp/dq/structured/bronze_to_silver_checkpoint")
+dbutils.widgets.text("quarantine_checkpoint", "/tmp/dq/structured/bronze_to_quarantine_checkpoint")
+dbutils.widgets.text("silver_table", "/tmp/dq/structured/silver_table")
+dbutils.widgets.text("quarantine_table", "/tmp/dq/structured/quarantine_table")
+
+# COMMAND ----------
+
+silver_checkpoint = dbutils.widgets.get("silver_checkpoint")
+quarantine_checkpoint = dbutils.widgets.get("quarantine_checkpoint")
+silver_table = dbutils.widgets.get("silver_table")
+quarantine_table = dbutils.widgets.get("quarantine_table")
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Bronze Stream
+# MAGIC
+# MAGIC This cell reads the NYC Taxi 2019 dataset from a Delta table as a streaming DataFrame, which will be used as the input for downstream data quality checks and transformations.
+
+# COMMAND ----------
+
+bronze_stream = (
+    spark.readStream
+    .format("delta")
+    .load("/databricks-datasets/delta-sharing/samples/nyctaxi_2019")
+)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Apply data quality checks and split the bronze stream into silver and quarantine DataFrames
+
+# COMMAND ----------
+
+from databricks.labs.dqx.engine import DQEngine
+from databricks.sdk import WorkspaceClient
+from pyspark.sql import DataFrame
+
+dq_engine = DQEngine(WorkspaceClient())
+
+silver_df, quarantine_df = dq_engine.apply_checks_by_metadata_and_split(bronze_stream, checks)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Write streaming DataFrames to Delta tables with checkpointing
+
+# COMMAND ----------
+
+PATH_PREFIXES = ("/", "dbfs:/", "s3://", "abfss://", "gs://")
+
+def write_stream(df, checkpoint_location, target):
+    writer = (
+        df.writeStream
+            .format("delta")
+            .outputMode("append")
+            .option("checkpointLocation", checkpoint_location)
+            .trigger(availableNow=True)
```

Comment:
would be good to have a comment here, that the processing will stop as soon as all data is processed